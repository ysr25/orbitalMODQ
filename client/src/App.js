import React, { Component } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import axios from "axios";
import "bootstrap/dist/css/bootstrap.min.css";

import Homepage from "./directories/Homepage";
import CreateReview from "./directories/CreateReview";
import EditReview from "./directories/EditReview";
import Register from "./directories/Register";
import LogIn from "./directories/LogIn";
import ViewReview from "./directories/ViewReview";
import Navbar from "./components/Navbar";
import EditUser from "./directories/EditUser";
import NotFoundPage from "./directories/NotFoundPage";

class App extends Component {
  constructor() {
    super();
    this.state = {
      loggedIn: false,
    };
  }

  api = (method, url, data = {}, params = {}) => {
    console.log("api call", method, url);
    return axios
      .request({
        method: method,
        url: "/api" + url,
        data: data,
        params: params,
        withCredentials: true,
      })
      .then((res) => {
        this.setState({ loggedIn: res.data.isLoggedIn });
        return res;
      });
  };

  updateUser = () => {
    this.api("get", "/").catch((err) => console.log(err));
  };

  render() {
    return (
      <Router>
        <div className="container">
          <Navbar
            updateUser={this.updateUser}
            loggedIn={this.state.loggedIn}
            api={this.api}
          />
          <br />
          <Switch>
            <Route
              exact
              path="/"
              render={(props) => <Homepage {...props} api={this.api} />}
            />
            <Route
              path="/reviews/new"
              render={(props) => <CreateReview {...props} api={this.api} />}
            />
            <Route
              path="/reviews/edit/:id"
              render={(props) => <EditReview {...props} api={this.api} />}
            />
            <Route
              path="/reviews/view/:id"
              render={(props) => (
                <ViewReview
                  {...props}
                  loggedIn={this.state.loggedIn}
                  api={this.api}
                />
              )}
            />
            <Route
              exact
              path="/users/signup"
              render={(props) => <Register {...props} api={this.api} />}
            />
            <Route
              path="/users/login"
              render={(props) => <LogIn {...props} api={this.api} />}
            />
            <Route
              exact
              path="/users/edit"
              render={(props) => <EditUser {...props} api={this.api} />}
            />
            <Route
              path="*"
              render={(props) => <NotFoundPage {...props} api={this.api} />}
            />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
