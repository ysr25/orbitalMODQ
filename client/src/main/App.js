import React, { Component } from 'react'
import { BrowserRouter as Router, Route, Link } from 'react-router-dom'
import axios from 'axios'
import 'bootstrap/dist/css/bootstrap.min.css'

import Homepage from './directories/Homepage'
import CreateModReview from './directories/CreateModReview'
import EditModReview from './directories/EditModReview'
import Registration from './directories/Registration'
import LoginPage from './directories/LoginPage'
import ViewPost from './directories/ViewPost'
import Navbar from './components/Navbar'
import EditUser from './directories/EditUser'
import logo from './logo.jpg'

class App extends Component {
  constructor () {
    super()
    this.state = {
      loggedIn: false
    }
  }

  api = (method, url, data = {}, params = {}) => {
    console.log('api call', method, url)
    return axios.request({
      method: method,
      url: '/api' + url,
      data: data,
      params: params,
      withCredentials: true
    })
      .then(res => {
        this.setState({ loggedIn: res.data.isLoggedIn })
        return res
      })
  }

  updateUser = () => {
    this.api('get', '/')
      .catch(err => console.log(err))
  }

  setShow = (show) => {
    this.setState({ show: false })
  }

  render () {
    return (
      <Router>
        <div className='container'>
          <nav className='navbar navbar-expand-lg navbar-light bg-light'>
            <Link to='/' className='navbar-brand' onClick={this.updateUser}>
              <img src={logo} width='50' height='50' alt='MODQ Logo' />
              MODQ
            </Link>
            <Navbar
              updateUser={this.updateUser}
              loggedIn={this.state.loggedIn}
              api={this.api}
            />
          </nav>
          <br />
          <Route
            exact path='/'
            render={props => <Homepage {...props} api={this.api} />}
          />
          <Route 
            path='/modreviews/newpost' 
            render={props => <CreateModReview {...props} api={this.api} />}
          />
          <Route
            path='/modreviews/edit/:id'
            render={props => <EditModReview {...props} api={this.api} />}
          />
          <Route
            path='/modreviews/view/:id'
            render={props => <ViewPost
              {...props}
              loggedIn={this.state.loggedIn}
              api={this.api}
            />}
          />
          <Route
            exact path='/users/signup'
            render={props => <Registration {...props} api={this.api} />}
          />
          <Route
            path='/users/login'
            render={props => <LoginPage {...props} api={this.api} />}
          />
          <Route
            exact path='/users/edit'
            render={props => <EditUser {...props} api={this.api} />}
          />
        </div>
      </Router>
    )
  }
}

export default App
