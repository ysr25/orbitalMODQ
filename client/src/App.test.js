import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import App from './App'

test('renders app', () => {
  // const { getByText } = render(<App />);
  // const linkElement = getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
  const tree = renderer
    .create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
