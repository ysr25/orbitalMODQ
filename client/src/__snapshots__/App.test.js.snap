// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<div
  className="container"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="MODQ Logo"
        height="50"
        src="logo.jpg"
        width="50"
      />
      MODQ
    </a>
    <div
      className="collpase navbar-collapse"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="navbar-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="navbar-item"
        >
          <a
            className="nav-link"
            href="/modreviews/newpost"
            onClick={[Function]}
          >
            Create A Post
          </a>
        </li>
        <li
          className="navbar-item"
        >
          <a
            className="nav-link"
            href="/users/signup"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
        <li
          className="navbar-item"
        >
          <a
            className="nav-link"
            href="/users/login"
            onClick={[Function]}
          >
            Log In
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <br />
  <div
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <div>
      <div
        className="btn-group btn-group-toggle"
        role="group"
      >
        <label
          className="btn active btn-outline-info"
          disabled={false}
        >
          <input
            autoComplete="off"
            checked={true}
            disabled={false}
            name="sort_by"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="datePosted"
          />
          Date posted
        </label>
        <label
          className="btn btn-outline-info"
          disabled={false}
        >
          <input
            autoComplete="off"
            checked={false}
            disabled={false}
            name="sort_by"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="dateEdited"
          />
          Date edited
        </label>
        <label
          className="btn btn-outline-info"
          disabled={false}
        >
          <input
            autoComplete="off"
            checked={false}
            disabled={false}
            name="sort_by"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value="upvotes"
          />
          Upvotes
        </label>
      </div>
       
      <div
        className="btn-group btn-group-toggle"
        role="group"
      >
        <label
          className="btn btn-outline-info"
          disabled={false}
        >
          <input
            autoComplete="off"
            checked={false}
            disabled={false}
            name="sort_by"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={-1}
          />
          Oldest to Latest Post
        </label>
        <label
          className="btn active btn-outline-info"
          disabled={false}
        >
          <input
            autoComplete="off"
            checked={true}
            disabled={false}
            name="sort_by"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={1}
          />
          Latest to Oldest Post
        </label>
      </div>
      <br />
      <br />
      <input
        autoComplete="off"
        className="form-control"
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        placeholder="Choose a module"
        value=""
      />
      <br />
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-row"
        >
          <div
            className="col"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Search reviews"
              required={true}
              value=""
            />
          </div>
          <div
            className="col"
          >
            <button
              className="btn btn-info"
              disabled={false}
              type="submit"
            >
              Search
            </button>
             
            
            <button
              className="btn btn-info"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Reset
            </button>
          </div>
        </div>
      </form>
    </div>
    <br />
  </div>
</div>
`;
